### PR TITLE
Use and configure Poetry for packaging

### PR DESCRIPTION
**Pull request by mmouchous-ledger**

### CODE REVIEW

File: laserstudio/__init__.py [hunk 1]
```diff
@@ -4,5 +4,5 @@
 Hardware evaluation tool developed by the Donjon
 """
 
-__version__ = "1.0.0dev3"
+__version__ = "0.0.0"
```
Comment (kingofpayne) [inline laserstudio/__init__.py:7]: Was expecting "1.0.0", is this correct?

File: laserstudio/__main__.py [hunk 2]
```diff
@@ -68,4 +69,8 @@
     win = LaserStudio(yaml_config)
     win.setWindowTitle(app.applicationDisplayName())
     win.show()
-    sys.exit(app.exec())
+    return app.exec()
```
Comment (kingofpayne) [inline laserstudio/__main__.py:72]: This does not forward error codes from Qt to the shell, like `sys.exit` does.
Maybe you want the caller of `main` to call `sys.exit` with the result returned from `main`?

File: laserstudio/__main__.py [hunk 3]
```diff
@@ -68,4 +69,8 @@
...
+
+
+if __name__ == "__main__":
+    main()
```
Comment (kingofpayne) [inline laserstudio/__main__.py:76]: Related to previous comment:
`sys.exit(main())` ?

**Review by kingofpayne — APPROVED** [review]

File: laserstudio/__main__.py (hunk 2 above)
Comment (mmouchous-ledger) [reply laserstudio/__main__.py:72]: When building with poetry, the entrypoints are generated with wrapping 'sys.exit', so main() should be called with sys.exit() wrapping, and it is indeed not the case in line 76.

File: laserstudio/__main__.py (hunk 3 above)
Comment (mmouchous-ledger) [reply laserstudio/__main__.py:76]: Yes.

File: laserstudio/__init__.py (hunk 1 above)
Comment (mmouchous-ledger) [reply laserstudio/__init__.py:7]: It is something automatically updated by poetry when publishing with the git tag version as "v1.0.0". Bumping will be done in main branch

**Pull request merged**